brambling/static/brambling/javascript/components/register.ractive.js

```diff
@@ -0,0 +1,252 @@
+                }
+            });
+        },
+        countdown: function() {
+            var thisObj = this,
+                startStr = thisObj.get('order.cart_start_time'),
+                cartTimeout = thisObj.get('event.cart_timeout');
+            if (!startStr || !cartTimeout) {
+                thisObj.set('countdown', undefined);
+                return;
+            }
+            var startDate = new Date(startStr),
+                now = new Date(),
+                elapsed = now - startDate,
+                remaining = cartTimeout * 60 * 1000 - elapsed;
+
+            if (remaining < 0) {
+                // cart has expired. show modal / clear cart.
+                thisObj.set('countdownExpired', true);
+                thisObj.loadOrder();
+                return;
+            }
+
+            var seconds_left = Math.floor(remaining / 1000),
+                countdown = {
+                    seconds: seconds_left % 60,
+                    minutes: Math.floor(seconds_left/60) % 60,
+                    hours: Math.floor(seconds_left/3600) % 24,
+                    days: Math.floor(seconds_left/86400)
+                };
+            thisObj.set('countdown', countdown);
+
+            setTimeout(function() {
+                thisObj.countdown();
+            }, 1000);
+        },
+
+        addToCart: function(item_option) {
+            var thisObj = this,
+                order = thisObj.get('order');
+            $.ajax({
+                url: thisObj.apiEndpoints['boughtitem'],
+                method: 'post',
+                data: {
+                    item_option: item_option.link,
+                    order: order.link
+                },
+                success: function (data) {
+                    thisObj.push('order.bought_items', data)
+                    thisObj.loadItems();
+                    thisObj.loadOrder();
+                }
+            })
+        },
+        removeFromCart: function(bought_item) {
+            var thisObj = this;
+            $.ajax({
+                url: bought_item.link,
+                method: 'delete',
+                data: {},
+                success: function(data) {
+                    var bought_items = thisObj.get('order.bought_items'),
+                        index = null;
+
+                    $.each(bought_items, function(idx, item) {
+                        if (item.link === bought_item.link) {
+                            index = idx;
+                            return false;
+                        }
+                    });
+                    if (index !== null) {
+                        thisObj.splice('order.bought_items', index, 1);
+                    }
+                    thisObj.loadItems();
+                    thisObj.loadOrder();
+                }
+            });
+        },
+        oninit: function (options) {
+            var thisObj = this;
+            thisObj.observe("discountCode", function (newValue, oldValue) {
```

Comment:
All this component talk has me wondering if this (discount handling) should be a separate component as well?